src/query/api/v1/handler/prometheus/remote/attribution.go

```diff
@@ -75,16 +75,18 @@ type promAttributionMetrics struct {
 }
 
 func (pam *promAttributionMetrics) match(labels []prompb.Label) bool {
```

Comment:
Pay attention to the following case. It returns true. I'm not sure it's intended.
```
labels = ["a", "a",]

pam.matchers = {"a": always_match, "b", xxx}
```
`matched` will be 2, because `"a"` matches twice.

Reply:
The assumption is prometheus time series won't have same labels in a single time series.